straxen/analyses/bokeh_waveform_plot.py

```diff
@@ -0,0 +1,488 @@
+import bokeh
+import bokeh.plotting as bklt
+import numpy as np
+import strax
+import straxen
+
+# Default legend and color options for unknow, S1 and S2
+COLORS = ('gray', 'blue', 'green')
+LEGENDS = ('Unknown', 'S1', 'S2')
+
+
+@straxen.mini_analysis(requires=('events', 'event_basics'))
+def event_display_interactive(events, to_pe, run_id, context, xenon1t=False):
+    """
+    Interactive event display for XENONnT. Plots detailed main/alt
+    S1/S2, bottom and top PMT hit pattern as well as all other peaks
+    in a given event.
+
+    Note:
+        How to use:
+
+        > straxen.event_display(context,
+        >                       run_id,
+        >                       time_range=(event['time'],
+        >                                   event['endtime'])
+        >                      )
+
+    Warning:
+        Raises an error if the user queries a time range which contains
+        more than a single event.
+
+    :return: bokeh.plotting.figure instance.
+    """
+    if len(events) != 1:
+        raise ValueError('The time range you specified contains more or'
+                         ' less than a single event. The event display '
+                         ' only works with individual events for now.')
+        
+       
+    peaks = context.get_array(run_id,
```

Comment:
I do not understand this comment, I would like to plot all peaks which are within the specified event. 

About the dependency thing, you convinced me not to put it into the mini analysis. If you would by accident query an entire run your memory would die. Hence I changed it into this indirect way as you did.

Reply:
Haha yeah, maybe I explained it badly, because in contrary to what you are doing, I was calling other minianalyses. Therefore the event display did not need to add the lower level stuff for the entire wf (I only loaded some peaklets directly in the S1 S2 time range which is much smaller). Do you see what I mean?

If you are afraid there will be too much data, you could add a ` warn_beyond_sec=1` argument to the minianalysis wrapper.